ggml-cuda.cu

```diff
@@ -9768,9 +9515,14 @@ static size_t ggml_backend_cuda_buffer_type_get_alloc_size(ggml_backend_buffer_t
 }
 
 static bool ggml_backend_cuda_buffer_type_supports_backend(ggml_backend_buffer_type_t buft, ggml_backend_t backend) {
-    return ggml_backend_is_cuda(backend);
+    if (!ggml_backend_is_cuda(backend)) {
+        return false;
+    }
 
-    UNUSED(buft);
+    int device = (int) (intptr_t) buft->context;
```

Comment:
`context` is a void pointer meant to be used by the backends to store whatever data they need. In this case, it is used to store the device of the buffer type. Since it is just an int, a cast from pointer to int used instead of allocating some struct and making it point to that.

Reply:
But doesn't this line cast from `int *` to `int`? Does this dereference the pointer or does it put convert the memory address to `int`?

Reply:
It uses the address as an int, the pointer is never dereferenced.